### PR TITLE
PWA-628 Search field bugs

### PR DESCRIPTION
**Pull request by ainarssondors**
-Fixed mobile bug where search results would overlap cart/login pages

CR & QA required before merge

### CODE REVIEW

File: src/app/component/Header/Header.container.js [hunk 1]
```diff
@@ -229,20 +251,21 @@ export class HeaderContainer extends NavigationAbstractContainer {
...
         if (!isMobile.any() && name === SEARCH) {
-            this.setState({ searchCriteria: '' });
-
-            hideActiveOverlay();
+            this.hideSearchOverlay();
             goToPreviousNavigationState();
         }
     }
 
+    goToDefaultHeaderState() {
```
Comment (alfredsgenkins) [inline src/app/component/Header/Header.container.js:262]: Maybe worth moving into abstract navigation? Sounds pretty logical to fallback onto default route for the pathname.

File: src/app/component/Header/Header.container.js [hunk 2]
```diff
@@ -229,20 +251,21 @@ export class HeaderContainer extends NavigationAbstractContainer {
...
+        const { setNavigationState } = this.props;
+        const state = this.getNavigationState(location.pathname);
```
Comment (alfredsgenkins) [inline src/app/component/Header/Header.container.js:264]: Why not to use the `location` from props?

File: src/app/component/Header/Header.component.js [hunk 3]
```diff
@@ -381,7 +391,12 @@ export default class Header extends NavigationAbstract {
                   mods={ { isVisible, type: 'minicart' } }
                 >
                     <button
-                      onClick={ onMinicartButtonClick }
+                      onClick={ () => {
+                          onMinicartButtonClick();
+                          if (name === CART) {
+                              onMinicartOutsideClick();
```
Comment (alfredsgenkins) [inline src/app/component/Header/Header.component.js:397]: Just make sure you really want to run: `onMinicartButtonClick();` and `onMinicartOutsideClick();` together if `name === CART`. Just pay attention, will it work with only one function called if `name === CART`?

File: src/app/component/Header/Header.style.scss [hunk 4]
```diff
@@ -81,6 +81,14 @@
             }
         }
 
+        &_cart_overlay {
+            .Header-Button_type_minicart {
+                @include after-mobile {
+                    --header-color: var(--header-desktop-button-color);
```
Comment (alfredsgenkins) [inline src/app/component/Header/Header.style.scss:87]: Hmm, just make sure the `var(--header-desktop-button-color);` is right var to use. Prefer more low-level if possible, i.e. `--primary-base-color`.

File: src/app/component/SearchField/SearchField.style.scss [hunk 5]
```diff
@@ -37,7 +37,6 @@
     }
 
     &_isActive {
-        --search-bar-width: 450px;
```
Comment (alfredsgenkins) [inline src/app/component/SearchField/SearchField.style.scss:40]: Does not this break tablet? I am not saying it must work on all screen-sizes, but just make sure it look ok on most screen-sizes.

File: src/app/component/SearchOverlay/SearchOverlay.style.scss [hunk 6]
```diff
@@ -39,6 +39,8 @@
     }
 
     &_isVisible {
+        z-index: 100;
```
Comment (alfredsgenkins) [inline src/app/component/SearchOverlay/SearchOverlay.style.scss:42]: This z-index I think is already used by some other elements, ensure they do not conflict.

File: src/app/component/Header/Header.container.js (hunk 1 above)
Comment (ainarssondors) [reply src/app/component/Header/Header.container.js]: Moved to abstract navigation.

File: src/app/component/Header/Header.container.js (hunk 2 above)
Comment (ainarssondors) [reply src/app/component/Header/Header.container.js]: now uses `navigationState` from props

File: src/app/component/Header/Header.style.scss (hunk 4 above)
Comment (ainarssondors) [reply src/app/component/Header/Header.style.scss]: minicart icon now uses `--primary-base-color`

File: src/app/component/SearchField/SearchField.style.scss (hunk 5 above)
Comment (ainarssondors) [reply src/app/component/SearchField/SearchField.style.scss:40]: Now the value is set to `30vw` so as not to overlap logo when approaching tablet size

File: src/app/component/SearchOverlay/SearchOverlay.style.scss (hunk 6 above)
Comment (ainarssondors) [reply src/app/component/SearchOverlay/SearchOverlay.style.scss:42]: does not overlap demo notice and navigation tabs. Did not test the install prompt, though.

**Pull request merged**